### ISSUE TITLE
Error: Module did not self-register.

### ISSUE THREAD

**Issue by SamuelMarks** [ranisalt/node-argon2#127]
### Before creating an issue, please be sure to:
- [x] Checkout to the latest version, **including submodules**
- [x] Try to find an *isolated* way to reproduce the behavior
- [x] Fill in all the blanks in the most specific way you can

### Steps to reproduce
1. `node -r argon2` on latest LTS (Node.js v8.11.1)

### Expected behaviour
Should work without error

### Actual behaviour
```
$ npm i argon2
> argon2@0.18.1 install /private/tmp/a/node_modules/argon2
> node-gyp rebuild

  CC(target) Release/obj.target/libargon2/argon2/src/opt.o
  CC(target) Release/obj.target/libargon2/argon2/src/argon2.o
  CC(target) Release/obj.target/libargon2/argon2/src/core.o
  CC(target) Release/obj.target/libargon2/argon2/src/blake2/blake2b.o
  CC(target) Release/obj.target/libargon2/argon2/src/thread.o
  CC(target) Release/obj.target/libargon2/argon2/src/encoding.o
  LIBTOOL-STATIC Release/argon2.a
  CC(target) Release/obj.target/argon2/src/argon2.o
  SOLINK_MODULE(target) Release/argon2.node

+ argon2@0.18.1
added 4 packages from 10 contributors in 4.752s
```
Then:
```
$ node -r argon2
/private/tmp/a/node_modules/bindings/bindings.js:88
        throw e
        ^

Error: Module did not self-register.
    at Object.Module._extensions..node (module.js:681:18)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at bindings (/private/tmp/a/node_modules/bindings/bindings.js:81:44)
    at Object.<anonymous> (/private/tmp/a/node_modules/argon2/argon2.js:3:37)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
```

### Environment
**Operating system**:
macOS High Sierra 10.13.4 (17E199)
**Node version:**
v8.11.1
**Compiler version:**
```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.1.0 (clang-902.0.39.1)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

---

I've tried upgrading to the latest git version and completely removing all node, npm and yarn from my OS (then just reinstalling Node.js LTS and npm 5.8). Error remains.

**Comment by ranisalt** [comment]
I can't reproduce, but it seems to be happening to Travis as well.

**Comment by SamuelMarks** [comment]
FYI: It works fine with v0.17.3.

**Comment by ranisalt** [comment]
This might be the dumbest behavior I've ever seen on my programming life. Bindings **requires** the main module file to be named `<module>_node.cpp`, it's not like I haven't indicated in code what was the module file. This is ludicrous.

**Comment by ranisalt** [comment]
Fixed on 0.18.2, hopefully :smiley:

**Comment by alexanderhorner** [comment]
This is still happening with nuxt3 hot reload
